### PR TITLE
Updating Focus Within

### PR DESCRIPTION
**Pull request by Antonio-Laguna**
This is part of #460 

This converts the plugin to use the scaffolding process we've created so docs are easier to maintain. Here is a quick summary of the changes made:

- Made so selectors now include the ready class `.js-focus-within`.
- Rewrote browser polyfill to be included within the postcss plugin rather than using JSX Tool's one
- Included browser testing.

This is basically a duplicate of `focus-visible` with some enhancements from blank given that in this case we do control the Polyfill now.

### CODE REVIEW

File: plugins/postcss-focus-visible/README.md [hunk 1]
```diff
@@ -59,10 +59,10 @@ instructions for:
 ### preserve
 
 The `preserve` option determines whether the original notation
-is preserved. By default, it is not preserved.
+is preserved. By default, it is preserved.
```
Comment (Antonio-Laguna) [inline plugins/postcss-focus-visible/README.md:62]: This was actually wrong as per https://github.com/csstools/postcss-plugins/blob/main/plugins/postcss-focus-visible/src/index.ts#L7

File: plugins/postcss-focus-visible/src/index.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ const creator: PluginCreator<pluginOptions> = (opts?: pluginOptions) => {
 
 				let containsPseudo = false;
 				selectorAST.walkPseudos((pseudo) => {
-					if (pseudo.value !== PSEUDO) {
+					if (pseudo.value.toLowerCase() !== PSEUDO) {
```
Comment (Antonio-Laguna) [inline plugins/postcss-focus-visible/src/index.ts:49]: Handling lowercase

**Comment by Antonio-Laguna** [comment]
@romainmenke I bet we should add now the `.js-blank-pseudo` to PostCSS Blank Pseudo too?

File: plugins/postcss-focus-visible/README.md [hunk 3]
```diff
@@ -98,14 +93,7 @@ postcssFocusVisible({ replaceWith: '[data-focus-visible-added]' })
...
 
-.foo {
-	color: blue;
-	color: red;
-}
-
-.baz {
-	color: green;
-}
+<example.preserve-true.expect.css>
```
Comment (romainmenke) [inline plugins/postcss-focus-visible/README.md:96]: I think the docs need a poke?

File: plugins/postcss-focus-within/src/browser.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+				activeElement = activeElement.parentNode;
+			}
+		}
+	};
+}
+
+export default function focusWithin(opts) {
+	// configuration
+	const options = Object.assign(
```
Comment (romainmenke) [inline plugins/postcss-focus-within/src/browser.js:40]: `Object.assign` doesn't have good legacy support.
I will do another pass to check for API usage.

**Review by romainmenke — COMMENTED** [review]
Awesome work!

File: plugins/postcss-focus-visible/README.md (hunk 3 above)
Comment (Antonio-Laguna) [reply plugins/postcss-focus-visible/README.md:99]: Nice catch!

File: plugins/postcss-focus-within/src/browser.js (hunk 4 above)
Comment (Antonio-Laguna) [reply plugins/postcss-focus-within/src/browser.js]: Thanks for flagging this. You're completely right! Thanks for putting me in the right path :)

**Review by romainmenke — APPROVED** [review]
LGTM!

> I bet we should add now the .js-blank-pseudo to PostCSS Blank Pseudo too?

Yeah, I think that would be best for consistency

**Pull request merged**